app/(home)/activities/page.tsx

```diff
@@ -1,10 +1,147 @@
+"use client";
+import { useState } from "react";
 import PageHero from "@/components/shared/PageHero";
-import React from "react";
+import { upcomingEvents, events } from "@/libs/constant/events.constant";
+import EventCarousel from "@/components/EventCarousel";
+
+// Define types for event items
+interface EventItem {
+  title: string;
+  image: string;
+  alt: string;
+  des: string;
+}
+
+// Define props for the Modal component
+interface ModalProps {
+  image?: string;
+  alt?: string;
+  onClose: () => void;
+}
+
+// Modal component
+const Modal: React.FC<ModalProps> = ({ image, alt, onClose }) => {
+  if (!image) return null;
+
+  return (
+    <div
+      className="fixed inset-0 bg-black bg-opacity-75 flex items-center justify-center z-50"
+      onClick={onClose}
+    >
+      <div className="relative bg-white p-4 rounded-lg">
+        <button
+          onClick={onClose}
+          className="absolute top-2 right-2 text-gray-600 hover:text-gray-800"
+        >
+          &times;
+        </button>
+        <img
+          src={image}
+          alt={alt}
+          className="w-full h-auto max-w-full max-h-screen object-contain"
+        />
+      </div>
+    </div>
+  );
+};
+
+const Activities: React.FC = () => {
+  const [selectedImage, setSelectedImage] = useState<{
+    image: string;
+    alt: string;
+  } | null>(null);
+
+  const handleImageClick = (image: string, alt: string) => {
+    setSelectedImage({ image, alt });
+  };
+
+  const closeModal = () => {
+    setSelectedImage(null);
+  };
 
-const Activities = () => {
   return (
-    <section className="px-3">
+    <section className="p-3">
       <PageHero pageHeroTitle="Activities & Events" />
+
+      {/* Upcoming section starts */}
+      <div className="max-w-full lg:max-w-[1500px] mx-auto">
+        <h1 className="text-green-500 md:text-3xl text-2xl font-semibold mt-6 mb-4">
+          Upcoming Events
+        </h1>
+        <div className="flex flex-col gap-5 md:flex-row md:flex-wrap">
+          {upcomingEvents.map((item: EventItem) => (
+            <div
+              key={item.title}
+              className="bg-white items-center p-3 rounded-lg shadow-lg flex flex-col md:flex-row cursor-pointer relative"
```

Comment:
Instead of shadows use only border prop , this makes the card a lot more cleaner